File: core/src/main/java/com/tuuzed/androidx/recyclerview/adapter/AbstractItemViewBinder.java
package com.tuuzed.androidx.recyclerview.adapter;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;

public abstract class AbstractItemViewBinder<T> implements ItemTypeViewBinder<T, CommonItemViewHolder> {

    @LayoutRes
    public abstract int getLayoutId();

    @NonNull
    @Override
    public CommonItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(getLayoutId(), parent, false);
        return new CommonItemViewHolder(itemView);
    }

}

File: adapter/src/main/java/com/tuuzed/recyclerview/adapter/AbstractItemViewBinder.kt
package com.tuuzed.recyclerview.adapter


import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.annotation.LayoutRes

abstract class AbstractItemViewBinder<in T> : ItemTypeViewBinder<T, CommonItemViewHolder> {

    @LayoutRes
    abstract fun getLayoutId(): Int

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CommonItemViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(getLayoutId(), parent, false)
        return CommonItemViewHolder(itemView)
    }

}